src/pdfnorm.jl

```diff
@@ -0,0 +1,50 @@
+"""
+    pdfL2norm(d::Distribution)
+
+Return the L2 norm of the probability distribution function `f(x)` of the distribution `d`:
+
+```math
+\\int_{S} f(x)^{2} \\mathrm{d} x
```

Comment:
I just noticed that this is not the L2 norm but rather the squared L2 norm. So I suggest to either change the name of the function or to change the values to the actual L2 norm:

```suggestion
\\big(\\int_{S} f(x)^{2} \\mathrm{d} x \\big)^{1/2}
```

Reply:
Yes, I wanted to ask about the name.  I believe what I'm computing here is what @ablaom and @fkiraly asked for in #806, so I'd rather change the name, I'm not really attached to it.  What do you suggest?

Reply:
I got the impression they were actually interested in the L2 norm (i.e. the square root of the integral), due to their comments and the use of `l2norm(..)^2` in https://github.com/JuliaStats/Distributions.jl/issues/806#issuecomment-490819682. But maybe they would be more interested in the squared L2 norm? It would also be easy to provide both if there is a need for the squared norm.

Reply:
@devmotion @giordano We are interested in the *square* of the l2 norm. This defines the Brier entropy, as given, for example on p 88 of https://www.ism.ac.jp/editsec/aism/pdf/059_1_0077.pdf

Reply:
Indeed, I think `brier_entropy` is as good name for this as any (need to multiply square of l2 norm by -1/2). Later, we will have the `tsallis_entropy` (so we can implement the Tsallis proper scoring rule), etc. 

In the doc-string we can refer to the David paper. (Of course, the definition depends on a choice of dominating measure, which will either be counting measure or Lebesgue measure.)

Reply:
The name `brier_entropy` might be a bit unfortunate though since there exist different versions of the Brier score and the associated entropy in literature. E.g., the definition by David with the counting measure is neither consistent with the original definition by [Brier](https://journals.ametsoc.org/view/journals/mwre/78/1/1520-0493_1950_078_0001_vofeit_2_0_co_2.xml) nor the definition by [Gneiting](https://viterbi-web.usc.edu/~shaddin/cs699fa17/docs/GR07.pdf). Maybe it would be clearer to just define the squared L2 norm here and then downstream packages that want to implement e.g. Brier entropy can choose whatever definition they want?

Reply:
The [`distr6`](https://alan-turing-institute.github.io/distr6/) R package, suggested as reference by Franz, uses the name [`pdfSquared2Norm`](https://www.rdocumentation.org/packages/distr6/versions/1.5.0/topics/pdfSquared2Norm).  A bit mouthful but accurate.  To not use camelCase this could be `pdfsquared2norm`?  (I personally prefer snake_case, but it doesn't seem to the convention here, as far as I can see)

Reply:
It seems there exists a `pdfPNorm` as well though: https://alan-turing-institute.github.io/distr6/reference/ExoticStatistics.html#method-pdfPNorm

I think `pdfsquaredL2norm` or `pdfsquared2norm` are reasonable choices. It is probably better to define the squared L2 norm and avoid `sqrt` and `^2` operations, and only define `pdfL2norm(d) = sqrt(pdfsquaredL2norm(d))` later if needed.